poo/design_patterns/en/OBSERVER_PATTERN.md

```diff
@@ -0,0 +1,232 @@
+
+Let's add these 2 methods :
+
+```java
+interface Observer {
+    void notify();
+}
+
+interface Observable {
+    void notifyObservers();
+}
+```
+
+Okey </br> Of course we must be able to link an **Observer** to an **Observable** element :
+
+```java
+interface Observer {
+    void notify();
+}
+
+interface Observable {
+    void notifyObservers();
+    void addObserver(Observer observer);
+}
+```
+
+Let's try to rewrite our alarm system making use of these 2 interfaces :
+
+```java
+class Alarm implements Observable {
+    
+    private final List<Observer> observers = new ArrayList<>();
+    
+    public void alert() {
+        notifyObservers();
+        this.speaker.startAlarm();
+    }
+
+    @Override
+    public void addObserver(Observer observer) {
+        observers.add(observer);
+    }
+    
+    @Override
+    public void notifyObservers() {
+        observers.foreach(Observer::notify);
+    }
+}
+
+class PoliceCallerComponent implements Observer {
+    
+    @Override 
+    public void notify() {
+        System.out.println("Police successfully called");
+    }
+}
+
+class DoorLockerComponent implements Observer {
+
+    @Override
+    public void notify() {
+        lockAllDoors();
+        System.out.print("The thief is blocked inside");
+    }
+
+    private void lockAllDoors() {
+        //...
+    }
+
+}
+
+//...
+```
+
+*I'm not going to write all of the systems, you understood the main idea.*
+
+You can't figure yourself what the method `notifyObservers()` will produce ?
+Go read the course about the [functional interfaces](https://github.com/readthedocs-fr/notions/blob/master/java/interfaces_fonctionnelles/en/FUNCTIONAL_INTERFACES.md) as well as the note about the [Method Referencing](https://github.com/readthedocs-fr/notions/blob/master/java/interfaces_fonctionnelles/en/FUNCTIONAL_INTERFACES.md#toujours-plus-court-toujours-plus-loin).
+
+This line will produce the same effect as the following for loop :
+       
+```java
+    for(Observer observer : observers) {
+        observer.notify();
+    }
+```
+
+We now have a code that perfectly suits the **Observer Pattern**. Let's add a control center for the realistic touch and it will surely please to our boss.
+
+```java
+class ControlCenter {
+
+    private static Alarm alarm;
+
+    public static void main(String[] args) {
+
+        alarm = new Alarm();
+
+        alarm.addObserver(new PoliceCallerComponent());
+        alarm.addObserver(new DoorLockerComponent());
+        //...
+
+       System.out.println("Alarm successfully initialized");
+    }
+
+}
+```
+
+Perfect !
+
+Now, as soon as the `alert()` method from the `Alarm` class gets called, each one of its **Observers** will get notified and will react in consequence.
+
+And as bonus, our code follows a realistic logic, it's not the alarm which knows its **Observers** but our **control center** which defines what will happen when our alarm goes on.
```

Comment:
```suggestion
And as a bonus, our code follows a realistic logic, since it's not the alarm that knows its **Observers** but rather our **control center**, which defines what will happen when our alarm occurs.
```